workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildContext.cs

```diff
@@ -14,60 +13,37 @@ public struct BuildContext
-                var targetConfigs = environmentConfig.BuildTargets
-                    .Where(t => t.Enabled && (buildTargetFilter?.Contains(t.Target) ?? true))
-                    .ToList();
-
-                // Filter out any deprecated targets
-                var supportedTargets = targetConfigs
-                    .Where(c => !c.Deprecated)
-                    .ToList();
-
-                // Which build targets are not supported by current install?
-                var missingTargets = supportedTargets
-                    .Where(c => !BuildSupportChecker.CanBuildTarget(c.Target))
-                    .ToList();
-
-                // Error on missing required build support
-                if (missingTargets.Any(c => c.Required))
+                if (environmentConfig == null)
                 {
-                    var targetNames = string.Join(", ", missingTargets
-                        .Where(c => c.Required)
-                        .Select(c => c.Target.ToString()));
-                    throw new BuildFailedException(
-                        $"Build failed for {workerType}. Cannot build for required ({targetNames}) because build support is not installed in the Unity Editor.");
+                    continue;
                 }
 
-                // Log builds we're skipping
-                if (missingTargets.Count > 0)
+                IEnumerable<BuildTargetConfig> supportedTargets;
+                try
                 {
-                    var targetNames = string.Join(", ", missingTargets.Select(c => c.Target.ToString()));
-                    Debug.LogWarning(
-                        $"Skipping ({targetNames}) because build support is not installed in the Unity Editor and the build target is not marked as 'Required'.");
-
-                    supportedTargets.RemoveAll(t => missingTargets.Contains(t));
+                    supportedTargets = environmentConfig.GetSupportedTargets(contextFilter);
+                }
+                catch (BuildFailedException exception)
+                {
+                    throw new BuildFailedException($"Build failed for {workerType}. {exception.Message}");
                 }
 
                 result.AddRange(supportedTargets.Select(targetConfig => new BuildContext
                 {
                     WorkerType = workerType,
-                    BuildEnvironment = buildEnvironment,
-                    ScriptingImplementation = scriptImplementation ?? targetConfig.ScriptingImplementation,
+                    BuildEnvironment = contextFilter.BuildEnvironment,
+                    ScriptingImplementation = contextFilter.ScriptImplementation ?? targetConfig.ScriptingImplementation,
                     BuildTargetConfig = targetConfig,
-                    IOSSdkVersion = (targetConfig.Target == BuildTarget.iOS) ? iosSdkVersion : null
+                    IOSSdkVersion = (targetConfig.Target == BuildTarget.iOS) ? contextFilter.IOSSdkVersion : null
```

Comment:
These make me feel `BuildContextFilter` is misnamed as this isn't filtering build targets, but actually applying certain settings to them.

Reply:
yeah the naming came from the `BuildTargetFilter` field in this

`BuildContextSettings`?

Reply:
Yeah sgtm